src/webgl/primitives.js

```diff
@@ -800,59 +800,110 @@ p5.RendererGL.prototype.triangle = function(args) {
 
-        this.vertices.push(new p5.Vector(_x, _y, 0));
-        this.uvs.push(_x, _y);
+   var gId = 'arc|' + x + "|" + y + "|" + width + "|" + 
+   height + "|" + start + "|" + stop + "|" + mode + "|" + detailX + "|";
+
+   if (!this.geometryInHash(gId))
+   {
+
+    var _arc = function()
+    {
+        this.strokeIndices = [];
+
+        for (var i = 0; i <= this.detailX; i++)
+        {
+            var u = i / this.detailX;
+            var theta = (stop - start) * u + start;
+
+            var _x = 0.5 + Math.cos(theta)/2;
+            var _y = 0.5 + Math.sin(theta)/2;
+
+            this.vertices.push(new p5.Vector(_x, _y, 0));
+            this.uvs.push([_x, _y]);
+
+           if (i < this.detailX && mode == PIE)
+           {                                               
+                this.faces.push([0, i + 1, i + 2]);
+                this.strokeIndices.push([i + 1, i + 2]);
+           }
+
+           if (i < this.detailX - 1 && (mode == CHORD || mode == OPEN || mode == 'ellipse'))
+           {
+                this.faces.push([0, i + 1, i + 2]);
+                this.strokeIndices.push([i + 1, i + 2]);
+           }
+
+        }
+
+        if (mode == PIE || mode == null){
+            this.vertices.unshift(new p5.Vector(0.5,0.5,0));
+            this.uvs.unshift([0.5,0.5,0]);
+            this.faces.push([0, this.vertices.length - 2, this.vertices.length - 1]);
+            this.strokeIndices.push([0 , 1]);
+            this.strokeIndices.push([0, this.vertices.length - 1]);
+        }
+
+        else if (mode == CHORD){
+            this.strokeIndices.push([0 , 1]);
+            this.strokeIndices.push([0, this.vertices.length - 1]);
+        }
+
+        else if (mode == OPEN || mode == 'ellipse') {
+            this.strokeIndices.push([0 , 1]);
+        }
 
-        this.faces.push([0, (i + 1) % this.detailX + 1, i + 1]);
-      }
     };
-    var ellipseGeom = new p5.Geometry(detailX, 1, _ellipse);
-    ellipseGeom.computeNormals();
+
+    var arcGeom = new p5.Geometry(detailX, 1, _arc);
+    arcGeom.computeNormals();
+
     if (detailX <= 50) {
-      ellipseGeom._makeTriangleEdges()._edgesToVertices();
-    } else {
-      console.log('Cannot stroke ellipse with more than 50 detailX');
+      arcGeom._makeTriangleEdges()._edgesToVertices(arcGeom);
+    } 
+    else {
+      console.log('Cannot stroke arc with more than 50 detailX');
     }
 
-    this.createBuffers(gId, ellipseGeom);
-  }
+    this.createBuffers(gId, arcGeom);
+
+   }
 
-  // only a single ellipse (of a given detail) is cached: a circle of
-  // _diameter_ 1 (radius 0.5).
-  //
-  // before rendering, this circle is squished (technical term ;)
-  // appropriately and moved to the required location.
   var uMVMatrix = this.uMVMatrix.copy();
+  
   try {
-    this.uMVMatrix.translate([x, y, 0]);
+    if (mode == 'ellipse'){
+        this.uMVMatrix.translate([x, y, 0]);
+    }
+    else {
+        this.uMVMatrix.translate([x - 15, y - 15, 0]);
```

Comment:
I think there's a problem with the placement of the center point for arcs. in the ellipse mode, the p5.prototype.ellipse adjusts the center point based on the current ellipse mode, while for arc this is done in p5.Renderer2D.prototype. this should probably be moved to p5.prototype in core/2d_primitives.js instead, so both 2d and 3d mode can use it. then I think we can lose this adjustment, which I don't think will work for all arc sizes.

Reply:
I moved the function arcModeAdjust from p5.Renderer2D.prototype to p5.prototype.arc. The problem still persists :( arc is not rendered on the expected position.

Reply:
Fixed with latest commit.